packages/react/src/components/OneFilterPicker/OneFilterPicker.tsx

```diff
@@ -108,7 +108,7 @@ const FiltersRoot = <Definition extends FiltersDefinition>({
   useEffect(() => {
     setLocalFiltersValue(value)
     // eslint-disable-next-line react-hooks/exhaustive-deps -- We deep compare the filters object
-  }, [JSON.stringify(filters)])
+  }, [JSON.stringify(filters), JSON.stringify(value)])
```

Comment:
What about :
```
const filters = useMemo(() => buildFilters(rawFilters), [rawFilters]);
const value = useMemo(() => calculateValue(rawValue), [rawValue]);

useEffect(() => {
  setLocalFiltersValue(value);
}, [filters, value]);
```

Of course `buildFilters` and `calculateValue` will be our own functions

Or maybe we can use [use-deep-compare-effect](https://github.com/kentcdodds/use-deep-compare-effect)

Reply:
oh i didnt know [use-deep-compare-effect](https://github.com/kentcdodds/use-deep-compare-effect). It is something we need a lot!!
I will install it a start to refactor out code